src/analysis/normed_space/star/basic.lean

```diff
@@ -174,6 +182,16 @@ norm_mul_coe_unitary A ⟨U, hU⟩
 
 end cstar_ring
 
+lemma self_adjoint.nnnorm_pow_two_pow [normed_ring E] [star_ring E] [cstar_ring E]
+  {x : E} (hx : x ∈ self_adjoint E) (n : ℕ) : ∥x ^ 2 ^ n∥₊ = ∥x∥₊ ^ (2 ^ n) :=
```

Comment:
I think it would be nice to have a version for `x : self_adjoint E` as well.

Reply:
note, I renamed this `nnnorm_pow_two_pow_of_self_adjoint`, and then called the `x : self_adjoint E` version `self_adjoint.nnnorm_pow_to_pow` so that we have dot notation. Also, no `coe` in the name because `self_adjoint E` already has a norm structure.